File: ecommerce-java/user-service/src/main/java/com/ecommerce/userservice/domain/UserEntity.java
package com.ecommerce.userservice.domain;

import com.ecommerce.userservice.dto.UserDto;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Table(name = "users")
@NoArgsConstructor
@AllArgsConstructor
@Getter
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, length = 50, unique = true)
    private String email;
    @Column(nullable = false, length = 50)
    private String name;
    @Column(nullable = false, unique = true)
    private String userId;
    @Column(nullable = false)
    private String encryptedPwd;

    public static UserEntity userDtoToUserEntity(UserDto userDto, String encryptedPwd){
        return new UserEntity(
                0L,
                userDto.getEmail(),
                userDto.getName(),
                UUID.randomUUID().toString(),
                encryptedPwd
        );
    }
}

File: ecommerce-kotlin/user-service/src/main/kotlin/com/example/userservice/domain/UserEntity.kt
package com.example.userservice.domain

import com.example.userservice.dto.UserDto
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "users")
data class UserEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,
    val email: String,
    val name: String,
    val userId: String,
    val encryptedPwd: String,
    @CreationTimestamp
    val createdAt: LocalDateTime?,
    @UpdateTimestamp
    val updatedAt: LocalDateTime?

) {
    constructor(userDto: UserDto, encryptedPwd: String)
            : this(
        0L, userDto.email, userDto.name, UUID.randomUUID().toString(), encryptedPwd, null, null
    )
}
